### PR TITLE
Licence field on framework form creating

### PR DESCRIPTION
**Pull request by lmsilvera**
This change is [Reviewable](https://reviewable.io/reviews/opensalt/opensalt/334)

### CODE REVIEW

**Comment by roverwolf** [comment]
@lmsilvera Please update the branch and fix the conflicts.

@brandonopened @jmarks 
Should at least one licence be added as a fixture so that the select box isn't empty initially?
Or should the select box not display if there are no licences available?

**Comment by brandonopened** [comment]
Is it too late to add this license then as a default available license
Title: Attribution 4.0 International 
https://creativecommons.org/licenses/by/4.0/legalcode

**Comment by lmsilvera** [comment]
@roverwolf Pull request updated with the changes requested. There are some issues on Codacy but I think we can ignore them.

File: app/DoctrineMigrations/Version20180201165909.php [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    {
+        $sql = <<<'xENDx'
+INSERT INTO ls_def_licence (identifier, uri, title, licence_text, updated_at)
+    SELECT * FROM (SELECT :uuid, :uri, :title, :licence_text, NOW()) AS tmp
+    WHERE NOT EXISTS (
+            SELECT title FROM ls_def_licence WHERE title = :title
+        ) LIMIT 1;
+xENDx;
+        $insertLicence = $this->connection->prepare($sql);
+        $uuid = Uuid::uuid5(Uuid::NAMESPACE_URL, "data:text/x-ref;src=EXTERNAL_LICENCE");
```
Comment (roverwolf) [inline app/DoctrineMigrations/Version20180201165909.php]: The text that the UUID is being generated from is quite generic, would probably be better to use the URL mentioned in the licence_text field to make it easy to add others if desired.

File: app/DoctrineMigrations/Version20180201165909.php [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    {
+        $sql = <<<'xENDx'
+INSERT INTO ls_def_licence (identifier, uri, title, licence_text, updated_at)
+    SELECT * FROM (SELECT :uuid, :uri, :title, :licence_text, NOW()) AS tmp
+    WHERE NOT EXISTS (
+            SELECT title FROM ls_def_licence WHERE title = :title
+        ) LIMIT 1;
+xENDx;
+        $insertLicence = $this->connection->prepare($sql);
+        $uuid = Uuid::uuid5(Uuid::NAMESPACE_URL, 'data:text/x-ref;src=EXTERNAL_LICENCE');
```
Comment (roverwolf) [inline app/DoctrineMigrations/Version20180201165909.php]: The text that the UUID is being generated from is quite generic, would probably be better to use the URL mentioned in the licence_text field to make it easy to add others if desired.

Comment (lmsilvera) [reply app/DoctrineMigrations/Version20180201165909.php]: Sure thing, I have already update it.

**Pull request merged**